src/examples/client/simpleStreamableHttp.ts

```
@@ -24,50 +25,79 @@ async function main(): Promise<void> {
   const transport = new StreamableHTTPClientTransport(
     new URL('http://localhost:3000/mcp')
   );
+  let supportsStandaloneSse = false;
 
   // Connect the client using the transport and initialize the server
   await client.connect(transport);
+  console.log('Connected to MCP server');
+  // Open a standalone SSE stream to receive server-initiated messages
+  console.log('Opening SSE stream to receive server notifications...');
+  try {
+    await transport.openSseStream();
+    supportsStandaloneSse = true;
+    console.log('SSE stream established successfully. Waiting for notifications...');
+  }
+  catch (error) {
+    console.error('Failed to open SSE stream:', error);
+  }
+
+  // Set up notification handlers for server-initiated messages
   client.setNotificationHandler(LoggingMessageNotificationSchema, (notification) => {
     console.log(`Notification received: ${notification.params.level} - ${notification.params.data}`);
   });
+  client.setNotificationHandler(ResourceListChangedNotificationSchema, async (_) => {
+    console.log(`Resource list changed notification received!`);
+    const resourcesRequest: ListResourcesRequest = {
+      method: 'resources/list',
+      params: {}
+    };
+    const resourcesResult = await client.request(resourcesRequest, ListResourcesResultSchema);
+    console.log('Available resources count:', resourcesResult.resources.length);
+  });
 
-
-  console.log('Connected to MCP server');
   // List available tools
-  const toolsRequest: ListToolsRequest = {
-    method: 'tools/list',
-    params: {}
-  };
-  const toolsResult = await client.request(toolsRequest, ListToolsResultSchema);
-  console.log('Available tools:', toolsResult.tools);
+  try {
```

This try-catch shouldn't be necessary if the server's capabilities are checked first.

---

hmm, it wasn't my experience, if a server dosn't define tools we fail hard here -- I'll add it to follow ups to check